### PR TITLE
[pickers] Use `useRtl` instead of `useTheme` to access direction

### PR DESCRIPTION
**Pull request by flaviendelangle**
`useRtl` has been added in `@mui/system` 5.15.13 and we support 5.15.15 and above so we are good.

____

I'm checking the approach with the core team before opening for review

## Follow up

We need to replace all the imports and use `createUseThemeProps` instead of `useThemeProps`

image

And do the same on the Tree View, which will be a lot easier

### CODE REVIEW

File: packages/x-date-pickers-pro/src/internals/utils/valueManagers.ts [hunk 1]
```diff
@@ -144,12 +144,12 @@ export const getRangeFieldValueManager = <TDate extends PickerValidDate>({
       ...dateRangeSections.endDate,
     ]);
   },
-  getV6InputValueFromSections: (sections, localizedDigits, isRTL) => {
+  getV6InputValueFromSections: (sections, localizedDigits, isRtl) => {
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/internals/utils/valueManagers.ts:147]: I unified to capitalization strategy of this variable with what the core is doing

**Review by LukasTy — APPROVED** [review]
LGTM if the Core approves of it. 👌 
Thanks for taking care of it. 🙏

**Comment by flaviendelangle** [comment]
It's still in draft because I did not finished removing it :laughing:

**Comment by flaviendelangle** [comment]
@LukasTy it's good for review
Here is the link of my discussion with Brijesh: https://mui-org.slack.com/archives/C0583QU37EF/p1717492276312499

**Review by LukasTy — APPROVED** [review]
LGTM. 👍 
Thank you for taking care of it! 🙏

**Pull request merged**